code/extensions/UserDefinedForm_ControllerFreshdeskExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+class UserDefinedForm_ControllerFreshdeskExtension extends Extension
+{
+
+    private static $dependencies = [
+        'freshdesk' => '%$FreshdeskAPI',
+    ];
+
+    /**
+     * @var FreshdeskAPI
+     */
+    public $freshdesk;
+
+    public function updateEmailData($emailData, $attachments)
+    {
+        if (!$this->owner->ExportToFreshdesk) {
+            return false;
+        }
+
+        if (!defined('FRESHDESK_API_BASEURL') || empty('FRESHDESK_API_BASEURL')) {
```

Comment:
This is not an appropriate place to check for this - should be bundled with FreshdeskAPI (I reckon FreshdeskAPI shouldn't even instantiate if these are not set).